html5validator/validator.py

```diff
@@ -150,8 +152,26 @@ def validate(self, files):
             regex = re.compile(i)
             e = [l for l in e if not regex.search(l)]
 
-        if stderr:
-            LOGGER.error(stderr)
+        if e:
+            LOGGER.debug(e)
+            if self.format == "text":
+                LOGGER.error(e)
+                return len(e)
+            elif self.format == "gnu":
+                LOGGER.error(e)
+                return len(e)
+            elif self.format == "json" and e[0] != u'{"messages":[]}':
+                LOGGER.error(e)
+                return len(e)
+            elif self.format == "xml" and len(e) > 4:
```

Comment:
Same here?

And then is it necessary to treat the different formatting options differently? I would prefer they all get handled the same way.

Reply:
I agree that treating the different format options is bad idea, I only did it because it was a primitive way to fix the formatting issues. 
I am currently working on the XML formatting but running into errors where not all parts are being removed.

Reply:
The latest commit treats all format types the same.